packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -478,155 +490,334 @@ export class ModularChangeFamily
-				const rebasedField = fieldKind.changeHandler.rebaser.rebase(
+				const rebasedChangeset = fieldKind.changeHandler.rebaser.rebase(
 					fieldChangeset,
-					taggedBaseChange,
-					(child, baseChild) =>
-						this.rebaseNodeChange(
+					makeAnonChange(baseChangeset),
+					(child, baseChild) => {
+						assert(
+							baseChild === undefined,
+							"This field should not have any base changes",
+						);
+						return this.rebaseNodeChange(
 							child,
-							{ revision, change: baseChild },
+							undefined,
 							genId,
 							crossFieldTable,
+							baseChanges,
+							fieldFilter,
 							revisionMetadata,
 							constraintState,
-						),
+						);
+					},
 					genId,
 					manager,
 					revisionMetadata,
 				);
-
 				const rebasedFieldChange: FieldChange = {
 					fieldKind: fieldKind.identifier,
-					change: brand(rebasedField),
-				};
-
-				const rebaseData: RebaseData = {
-					fieldChange: rebasedFieldChange,
-					baseChange: taggedBaseChange,
+					change: brand(rebasedChangeset),
 				};
-
-				addFieldData(manager, rebaseData);
 				rebasedFields.set(field, rebasedFieldChange);
+				addFieldData(manager, rebasedChangeset);
 			}
 		}
 
 		return rebasedFields;
 	}
 
 	private rebaseNodeChange(
-		change: NodeChangeset,
-		over: TaggedChange<NodeChangeset>,
+		change: NodeChangeset | undefined,
+		over: TaggedChange<NodeChangeset> | undefined,
 		genId: IdAllocator,
-		crossFieldTable: CrossFieldTable<RebaseData>,
+		crossFieldTable: RebaseTable,
+		parentField: FieldChange | undefined,
+		fieldFilter: (baseChange: FieldChange, newChange: FieldChange | undefined) => boolean,
 		revisionMetadata: RevisionMetadataSource,
 		constraintState: ConstraintState,
-	): NodeChangeset {
-		const fieldChanges =
-			change.fieldChanges === undefined || over.change.fieldChanges === undefined
-				? change.fieldChanges
-				: this.rebaseFieldMap(
-						change.fieldChanges,
-						{
-							...over,
-							change: over.change.fieldChanges,
-						},
-						genId,
-						crossFieldTable,
-						revisionMetadata,
-						constraintState,
-				  );
+	): NodeChangeset | undefined {
+		if (change === undefined && over?.change?.fieldChanges === undefined) {
+			return undefined;
+		}
 
-		const changeSet: NodeChangeset = {
-			...change,
-			fieldChanges,
-		};
+		if (over?.change?.fieldChanges !== undefined && parentField !== undefined) {
+			crossFieldTable.baseMapToParentField.set(over.change.fieldChanges, parentField);
+		}
+
+		const baseMap: TaggedChange<FieldChangeMap> =
+			over?.change?.fieldChanges !== undefined
+				? {
+						...over,
+						change: over.change.fieldChanges,
+				  }
+				: makeAnonChange(new Map());
+
+		const fieldChanges = this.rebaseFieldMap(
+			change?.fieldChanges ?? new Map(),
+			baseMap,
+			genId,
+			crossFieldTable,
+			fieldFilter,
+			revisionMetadata,
+			constraintState,
+		);
+
+		const rebasedChange: NodeChangeset = {};
```

Comment:
We should return undefined here if `change === undefined && fieldChanges.size === 0`.